truss/templates/control/control/helpers/inference_server_controller.py

```diff
@@ -55,14 +61,19 @@ def apply_patch(self, patch_request):
                 return
 
             if req_prev_hash != self._current_running_hash:
-                raise ValueError(
+                raise InadmissiblePatch(
                     f"Unable to apply patch: expected prev hash \
                     to be {self._current_running_hash} but found {req_prev_hash}"
                 )
 
-            patches = [
-                Patch.from_dict(patch_dict) for patch_dict in patch_request["patches"]
-            ]
+            try:
+                patches = [
+                    Patch.from_dict(patch_dict)
+                    for patch_dict in patch_request["patches"]
+                ]
+            except Exception as exc:
```

Comment:
Can we get any more specific for this exception?  What do we expect to throw here?

Reply:
afaict this should be KeyError, let me use that. There is a risk of accidentally forgetting about an exception type that can be thrown, but perhaps it's low here.

Reply:
Is the case this could happen when there is a version mismatch between the patch client and the patch server implementation?

Reply:
Makes sense. Yeah a KeyError might be more straightforward to the reader here.  A couple options that are also good alternatives:
* Having `from_dict` or one of its parents through a more specialized exception that this is a version mismatch issue
* Track compatible versions statically and reject the request early if not within supported client versions. Would require passing client truss version as a header